### PR TITLE
Forbidden Cities Editorial 

### PR DESCRIPTION
**Pull request by brebenelmihnea**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: content/6_Advanced/BCC_2CC.mdx [hunk 1]
```diff
@@ -178,14 +178,165 @@ struct TwoEdgeCC {
 
 <Problems problems="bccSam" />
 
-note that BCCs contain EDGES not VERTICES
+An important observation is that if you can't go from node `a` to node `b` without passing through node `c`, it means that node `c` is a critical node (articulation point).
+Node `c` can split `a` and `b` into 2 different components if removed. This makes us think about biconnected components.
```
Comment (SansPapyrus683) [inline content/6_Advanced/BCC_2CC.mdx:182]: use latex

File: content/6_Advanced/BCC_2CC.mdx [hunk 2]
```diff
@@ -178,14 +178,165 @@ struct TwoEdgeCC {
...
 
+
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+vector<vector<int>> biconnected_components(vector<vector<int>> &g,
```
Comment (SansPapyrus683) [inline content/6_Advanced/BCC_2CC.mdx:199]: comments comments comments

File: content/6_Advanced/BCC_2CC.mdx [hunk 3]
```diff
@@ -178,14 +178,165 @@ struct TwoEdgeCC {
...
+                                           vector<bool> &is_cutpoint,
+                                           vector<int> &id) {
+	int n = (int)g.size();
+	vector<vector<int>> comps;
+	vector<int> stk, num(n), low(n);
```
Comment (SansPapyrus683) [inline content/6_Advanced/BCC_2CC.mdx:204]: separate lines

File: content/6_Advanced/BCC_2CC.mdx [hunk 4]
```diff
@@ -178,14 +178,165 @@ struct TwoEdgeCC {
...
+	};
+
+	dfs(0, -1);
+
+	auto find_lca = [&](int x, int y) {
+		if (depth[x] < depth[y]) swap(x, y);
+		int diff = depth[x] - depth[y];
+		for (int i = LOGMAX; i >= 0; i--)
+			if ((1 << i) & diff) x = up[x][i];
+		if (x == y) return x;
```
Comment (SansPapyrus683) [inline content/6_Advanced/BCC_2CC.mdx:302]: braces braces braces

File: content/6_Advanced/BCC_2CC.mdx [hunk 5]
```diff
@@ -178,14 +178,165 @@ struct TwoEdgeCC {
...
+		int lca2 = find_lca(y, z);
+
+		if (lca == z || (lca1 == lca && lca2 == z) ||
+		    (lca2 == lca && lca1 == z)) {
+			return true;
+		}
+		return false;
+	};
+
+	while (q--) {
```
Comment (SansPapyrus683) [inline content/6_Advanced/BCC_2CC.mdx:324]: use a for loop

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/BCC_2CC.mdx (hunk 5 above)
Comment (brebenelmihnea) [reply content/6_Advanced/BCC_2CC.mdx:324]: Uff

File: content/6_Advanced/BCC_2CC.mdx (hunk 2 above)
Comment (brebenelmihnea) [reply content/6_Advanced/BCC_2CC.mdx]: Alright

File: content/6_Advanced/BCC_2CC.mdx (hunk 3 above)
Comment (brebenelmihnea) [reply content/6_Advanced/BCC_2CC.mdx:204]: Done

File: content/6_Advanced/BCC_2CC.mdx (hunk 4 above)
Comment (brebenelmihnea) [reply content/6_Advanced/BCC_2CC.mdx]: ok you maniac

File: content/6_Advanced/BCC_2CC.mdx (hunk 1 above)
Comment (brebenelmihnea) [reply content/6_Advanced/BCC_2CC.mdx:182]: What do you mean ?

Comment (SansPapyrus683) [reply content/6_Advanced/BCC_2CC.mdx:182]: dollar signs instead of the inline code thing

File: content/6_Advanced/BCC_2CC.mdx [hunk 6]
```diff
@@ -178,14 +178,173 @@ struct TwoEdgeCC {
...
+	g.resize(n);
+	for (int i = 1; i <= m; i++) {
+		int x, y;
+		cin >> x >> y;
+		x--, y--;
+		g[x].push_back(y);
+		g[y].push_back(x);
+	}
+
+	const int LOGMAX = (int)ceil(log2(n));
```
Comment (ryanchou-dev) [inline content/6_Advanced/BCC_2CC.mdx:279]: move this up with declaration

File: content/6_Advanced/BCC_2CC.mdx [hunk 7]
```diff
@@ -178,14 +178,173 @@ struct TwoEdgeCC {
...
+	return build_tree();
+}
+
+int main() {
+	int n, m, q;
+	cin >> n >> m >> q;
+
+	vector<vector<int>> g;
+	g.resize(n);
+	for (int i = 1; i <= m; i++) {
```
Comment (ryanchou-dev) [inline content/6_Advanced/BCC_2CC.mdx:271]: 0-index loop

(optional nitpick)

File: content/6_Advanced/BCC_2CC.mdx [hunk 8]
```diff
@@ -178,14 +178,173 @@ struct TwoEdgeCC {
...
+	auto on_path = [&](int x, int y, int z) {
+		int lca = find_lca(x, y);
+		int lca1 = find_lca(x, z);
+		int lca2 = find_lca(y, z);
+
+		if (lca == z || (lca1 == lca && lca2 == z) ||
+		    (lca2 == lca && lca1 == z)) {
+			return true;
+		}
+		return false;
```
Comment (ryanchou-dev) [inline content/6_Advanced/BCC_2CC.mdx:329]: move this before dfs call

File: content/6_Advanced/BCC_2CC.mdx (hunk 6 above)
Comment (brebenelmihnea) [reply content/6_Advanced/BCC_2CC.mdx:279]: OK

File: content/6_Advanced/BCC_2CC.mdx (hunk 8 above)
Comment (brebenelmihnea) [reply content/6_Advanced/BCC_2CC.mdx:329]: Alright

File: content/6_Advanced/BCC_2CC.mdx (hunk 1 above)
Comment (brebenelmihnea) [reply content/6_Advanced/BCC_2CC.mdx]: Alright done

File: content/6_Advanced/BCC_2CC.mdx [hunk 9]
```diff
@@ -178,14 +177,175 @@ struct TwoEdgeCC {
...
+	dfs(0, -1, timer);
+	id.resize(n);
+
+	// Build the block-cut tree
+	function<vector<vector<int>>()> build_tree = [&]() {
+		vector<vector<int>> t(1);
+		int cnt = 0;
+		for (int node = 0; node < n; node++) {
+			if (is_cutpoint[node]) {
+				id[node] = cnt++;
```
Comment (SansPapyrus683) [inline content/6_Advanced/BCC_2CC.mdx:244]: rename `cnt` to smth more informative

Comment (brebenelmihnea) [reply content/6_Advanced/BCC_2CC.mdx]: Ok renamed

**Review by SansPapyrus683 — APPROVED** [review]
this is in advanced so idt it should matter too much

**Comment by brebenelmihnea** [comment]
> this is in advanced so idt it should matter too much

Bruh

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**